src/App.tsx

```diff
@@ -1,223 +1,207 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Error } from './components/Error';
+import { Filter } from './components/Filter';
+import { getTodos, createTodo, deleteTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoItem } from './components/TodoItem';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortFilter, setSortFilter] = useState('all');
+  const [visibelTodos, setVisibelTodos] = useState<Todo[]>([...todos]);
+  const [completedTodos, setCompletedTodos]
+  = useState<Todo[]>([...todos].filter(todo => todo.completed));
+
+  const [isError, setError] = useState(false);
+  const [messageError, setMessageError] = useState('');
+  const [newTodoTitle, setTitle] = useState('');
+  const [activeTodoId, setActiveTodoId] = useState([0]);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then(response => {
+          setTodos(response);
+        })
+        .catch(() => {
+          setError(true);
+          setMessageError('Todos from server were not gotten');
+        });
+    }
   }, []);
 
+  useEffect(() => {
+    setVisibelTodos(() => (
+      todos.filter(todo => {
+        switch (sortFilter) {
+          case 'active':
```

Comment:
Do not rely on the unknown string, make constants for this.
```
const FILTERS = {
  all: ‘all’,
  completed: ‘completed’,
  active: ‘active’,
};
```

Reply:
done